### PR TITLE
Green Belt

### PR DESCRIPTION
**Pull request by mvollmary**

### CODE REVIEW

**Review by StefanDziwok — APPROVED** [review]

**Review by merschs — CHANGES_REQUESTED** [review]
I think we should finish green/environment-dependent-configuration-parameters.md before merging. Currently, the assessment is still work in progress.

**Comment by mvollmary** [comment]
This PR is still in progress. I only created it so we have this topic in our project board. I converted it to a draft (didn't know about this feature :tada:)

File: green/ensure-only-using-components-which-are-under-active-development.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Ensure only using components which are under active development
+
+Ensure that you only use software components (e.g., libraries, applications) which are under active development. This also includes migrating software **before** it reaches end of life / support.
+
+## Benefits
+
+- All software components are up-to-date and receive security patches for discovered vulnerabilities.
+- Modern technologies help to minimize your attack surface.
```
Comment (merschs) [inline green/ensure-only-using-components-which-are-under-active-development.md:8]: what does modern mean? Is here "up-to-date" meant?

File: green/environment-dependent-configuration-parameters.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+## Assessment
+
+- Show your configuration files for all your different environments like dev, test and production
+  - URL specification for different environments
+  - Configuration parameters for backend / database systems
+
+## Related Activities
+
+- [Same Artifact for Environments](../orange/same-artifact-for-environments.md)
```
Comment (merschs) [inline green/environment-dependent-configuration-parameters.md:20]: I think I would like it to mark activities which are not of this belt, e.g., "Orange Belt: Same Artificat ..."

File: green/integration-of-vulnerability-issues-into-the-development-process.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Integration of vulnerability issues into the development process
```
Comment (merschs) [inline green/integration-of-vulnerability-issues-into-the-development-process.md:1]:
```suggestion
# Integration of Vulnerability Issues into the Development Process
```

File: green/integration-of-vulnerability-issues-into-the-development-process.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+- Enables priorization of security-relevant issues for the product owner.
+
+## Assessment
+
+- Verify that issues exist for all relevant vulnerabilities.
+- Verify that vulnerabilities from all sources are included.
+
+## Related Activities
+
+- [Treatment of vulnerabilities with severity high or higher](treatment-of-vulnerabilities-with-severity-high-or-higher.md)
```
Comment (merschs) [inline green/integration-of-vulnerability-issues-into-the-development-process.md:17]:
```suggestion
- [Treatment of Vulnerabilities with Severity High or Higher](treatment-of-vulnerabilities-with-severity-high-or-higher.md)
```

File: green/environment-dependent-configuration-parameters.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Environment-dependent configuration parameters
```
Comment (merschs) [inline green/environment-dependent-configuration-parameters.md:1]:
```suggestion
# Environment-Dependent Configuration Parameters
```

File: green/ensure-only-using-components-which-are-under-active-development.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
+# Ensure only using components which are under active development
```
Comment (merschs) [inline green/ensure-only-using-components-which-are-under-active-development.md:1]:
```suggestion
# Ensure Only Using Components Which Are Under Active Development
```

File: green/README.md [hunk 7]
```diff
@@ -1,5 +1,15 @@
...
+- [Prevent Secrets in Source Code](prevent-secrets-in-source-code.md)
+- [Environment-dependent configuration parameters](environment-dependent-configuration-parameters.md)
+- [Integration of vulnerability issues into the development process](integration-of-vulnerability-issues-into-the-development-process.md)
+- [Static analysis for all self-written components](static-analysis-for-all-self-written-components.md)
+- [Treatment of vulnerabilities with severity high or higher](treatment-of-vulnerabilities-with-severity-high-or-higher.md)
 - [Test of Container Images Regarding Known Vulnerabilities](test-of-container-images-regarding-known-vulnerabilities.md)
+- [Test security configuration of cloud environments](test-security-configuration-of-cloud-environments.md)
+- [Test security configuration of infrastructure](test-security-configuration-of-infrastructure.md)
+- [Nightly Dependency Version Upgrade](nightly-dependency-version-upgrade.md)
+- [Ensure only using components which are under active development](ensure-only-using-components-which-are-under-active-development.md)
```
Comment (merschs) [inline green/README.md:15]:
```suggestion
- [Ensure Only Using Components Which Are Under Active Development](ensure-only-using-components-which-are-under-active-development.md)
```

File: green/README.md [hunk 8]
```diff
@@ -1,5 +1,15 @@
-# 4 - Green Belt [work in progress]
+# 4 - Green Belt
 
 The green belt introduces static code analysis as well as first activities ensuring security during the lifecycle of your software. It is also the first belt introducing topics regarding infrastructure, configuration, and credentials.
 
+- [Test for Default Credentials](test-for-default-credentials.md)
+- [Prevent Secrets in Source Code](prevent-secrets-in-source-code.md)
+- [Environment-dependent configuration parameters](environment-dependent-configuration-parameters.md)
```
Comment (merschs) [inline green/README.md:7]:
```suggestion
- [Environment-Dependent Configuration Parameters](environment-dependent-configuration-parameters.md)
```

File: green/static-analysis-for-all-self-written-components.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
+# Static analysis for all self-written components
```
Comment (merschs) [inline green/static-analysis-for-all-self-written-components.md:1]:
```suggestion
# Static Analysis for All Self-Written Components
```

File: green/README.md [hunk 10]
```diff
@@ -1,5 +1,15 @@
...
+- [Integration of vulnerability issues into the development process](integration-of-vulnerability-issues-into-the-development-process.md)
+- [Static analysis for all self-written components](static-analysis-for-all-self-written-components.md)
```
Comment (merschs) [inline green/README.md:9]:
```suggestion
- [Static Analysis for All Self-Written Components](static-analysis-for-all-self-written-components.md)
```

File: green/test-security-configuration-of-cloud-environments.md [hunk 11]
```diff
@@ -0,0 +1,25 @@
+# Test security configuration of cloud environments
```
Comment (merschs) [inline green/test-security-configuration-of-cloud-environments.md:1]:
```suggestion
# Test Security Configuration of Cloud Environments
```

File: green/README.md [hunk 12]
```diff
@@ -1,5 +1,15 @@
...
+- [Treatment of vulnerabilities with severity high or higher](treatment-of-vulnerabilities-with-severity-high-or-higher.md)
 - [Test of Container Images Regarding Known Vulnerabilities](test-of-container-images-regarding-known-vulnerabilities.md)
+- [Test security configuration of cloud environments](test-security-configuration-of-cloud-environments.md)
```
Comment (merschs) [inline green/README.md:12]:
```suggestion
- [Test Security Configuration of Cloud Environments](test-security-configuration-of-cloud-environments.md)
```

File: green/test-security-configuration-of-infrastructure.md [hunk 13]
```diff
@@ -0,0 +1,28 @@
+# Test security configuration of infrastructure
```
Comment (merschs) [inline green/test-security-configuration-of-infrastructure.md:1]:
```suggestion
# Test Security Configuration of Infrastructure
```

File: green/README.md [hunk 14]
```diff
@@ -1,5 +1,15 @@
...
+- [Test security configuration of infrastructure](test-security-configuration-of-infrastructure.md)
```
Comment (merschs) [inline green/README.md:13]:
```suggestion
- [Test Security Configuration of Infrastructure](test-security-configuration-of-infrastructure.md)
```

File: green/treatment-of-vulnerabilities-with-severity-high-or-higher.md [hunk 15]
```diff
@@ -0,0 +1,27 @@
+# Treatment of vulnerabilities with severity high or higher
```
Comment (merschs) [inline green/treatment-of-vulnerabilities-with-severity-high-or-higher.md:1]:
```suggestion
# Treatment of Vulnerabilities With Severity High or Higher
```

File: green/README.md [hunk 16]
```diff
@@ -1,5 +1,15 @@
...
+# 4 - Green Belt
 
 The green belt introduces static code analysis as well as first activities ensuring security during the lifecycle of your software. It is also the first belt introducing topics regarding infrastructure, configuration, and credentials.
 
+- [Test for Default Credentials](test-for-default-credentials.md)
+- [Prevent Secrets in Source Code](prevent-secrets-in-source-code.md)
+- [Environment-dependent configuration parameters](environment-dependent-configuration-parameters.md)
+- [Integration of vulnerability issues into the development process](integration-of-vulnerability-issues-into-the-development-process.md)
+- [Static analysis for all self-written components](static-analysis-for-all-self-written-components.md)
+- [Treatment of vulnerabilities with severity high or higher](treatment-of-vulnerabilities-with-severity-high-or-higher.md)
```
Comment (merschs) [inline green/README.md:10]:
```suggestion
- [Treatment of Vulnerabilities With Severity High or Higher](treatment-of-vulnerabilities-with-severity-high-or-higher.md)
```

File: green/treatment-of-vulnerabilities-with-severity-high-or-higher.md [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+
+Focus on fixing the most relevant vulnerabilities without being overwhelmed of the potentially great number of low-impact findings.
+
+## Benefits
+
+- Prioritized treatment of most severe vulnerabilities.
+- The software is free of high-impact vulnerabilities, which decreases the chance of denial of service attacks and lost or manipulation of confidential data.
```
Comment (merschs) [inline green/treatment-of-vulnerabilities-with-severity-high-or-higher.md:8]:
```suggestion
- The software is free of high-impact vulnerabilities, which decreases the chance of successful denial of service attacks and loss or manipulation of confidential data.
```

File: green/treatment-of-vulnerabilities-with-severity-high-or-higher.md [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+
+## Assessment
+
+- Show the fix for the latest *n* high criticality findings.
```
Comment (merschs) [inline green/treatment-of-vulnerabilities-with-severity-high-or-higher.md:12]:
```suggestion
- Show the fix for the latest *n* high severity findings.
```

File: green/treatment-of-vulnerabilities-with-severity-high-or-higher.md [hunk 19]
```diff
@@ -0,0 +1,27 @@
...
+
+## Related Activities
+
+- [False Positive Treatment](../yellow/false-positive-treatment.md)
+- [Test of Containers Regarding Known Vulnerabilities](../green/test-of-container-images-regarding-known-vulnerabilities.md)
+- [Test of Components Regarding Known Vulnerabilities](../yellow/test-of-components-regarding-known-vulnerabilities.md)
+- [Unit Tests for Security Features](../yellow/unit-tests-for-security-features.md)
+- [Test of Client Side Components With Known Vulnerabilities](../orange/test-of-client-side-components-with-known-vulnerabilities.md)
+- [Nightly Test of Libraries With Known Vulnerabilities](../orange/nightly-test-of-libraries-with-known-vulnerabilities.md)
+- [Static analysis for all self-written components](static-analysis-for-all-self-written-components.md)
```
Comment (merschs) [inline green/treatment-of-vulnerabilities-with-severity-high-or-higher.md:23]:
```suggestion
- [Static Analysis for All Self-Written Components](static-analysis-for-all-self-written-components.md)
```

File: green/README.md [hunk 20]
```diff
@@ -1,5 +1,15 @@
-# 4 - Green Belt [work in progress]
+# 4 - Green Belt
 
 The green belt introduces static code analysis as well as first activities ensuring security during the lifecycle of your software. It is also the first belt introducing topics regarding infrastructure, configuration, and credentials.
 
+- [Test for Default Credentials](test-for-default-credentials.md)
+- [Prevent Secrets in Source Code](prevent-secrets-in-source-code.md)
+- [Environment-dependent configuration parameters](environment-dependent-configuration-parameters.md)
+- [Integration of vulnerability issues into the development process](integration-of-vulnerability-issues-into-the-development-process.md)
```
Comment (merschs) [inline green/README.md:8]:
```suggestion
- [Integration of Vulnerability Issues into the Development Process](integration-of-vulnerability-issues-into-the-development-process.md)
```

**Review by merschs — DISMISSED** [review]
some minor remarks and wording regarding high severity fixes
+ headling capitalization

**Review by TimSro — APPROVED** [review]

**Pull request merged**